python/tests/xir/test_parser.py

```diff
@@ -63,6 +63,39 @@ def test_options_lists(self, key, val):
         assert key in irprog.options
         assert irprog.options[key] == val
 
+    @pytest.mark.parametrize(
+        "script, adjoint",
+        [
+            ("adjoint ry(2.4) | [2];", True),
+            ("adjoint adjoint ry(2.4) | [2];", False),
+            ("adjoint ctrl[0, 1] adjoint ry(2.4) | [2];", False),
+            ("adjoint adjoint ctrl[0, 1] adjoint ry(2.4) | [2];", True),
+            ("ry(2.4) | [2];", False),
+        ],
+    )
+    def test_adjoint_modifier(self, script, adjoint):
+        """Test that adjoint modifier for gate statements works correctly"""
+        irprog = parse_script(script)
+
+        assert irprog.statements[0].is_adjoint is adjoint
+
+    @pytest.mark.parametrize(
+        "script, ctrl_wires",
+        [
+            ("ctrl[0, 1] ry(2.4) | [2];", (0, 1)),
+            ("ctrl[0, 1] ctrl[3] rz(4.2) | [2];", (0, 1, 3)),
+            ("ctrl[0, 1] adjoint ctrl[3] rx(3.1) | [4];", (0, 1, 3)),
+            ("ry(2.4) | [2];", tuple()),
```

Comment:
Just `()` works too:
```suggestion
            ("ry(2.4) | [2];", ()),
```